src/Framework/PaymentGateways/Traits/HandleHttpResponses.php

```diff
@@ -60,4 +61,25 @@ public function handleExceptionResponse(\Exception $exception, string $message)
         give_set_error('PaymentGatewayException', $message);
         give_send_back_to_checkout();
     }
+
+    /**
+     * This checks the server content-type to determine how to respond.
+     *
+     * v2 GiveWP forms uses content-type application/x-www-form-urlencoded
+     * v3 GiveWP forms uses content-type application/json and/or multipart/form-data
+     *
+     * @unreleased
+     *
+     * @return bool
+     */
+    protected function shouldSendAsJson(): bool
+    {
+        if (!isset($_SERVER['CONTENT_TYPE'])) {
+            return false;
+        }
+
+        $contentType = isset($_SERVER['CONTENT_TYPE']);
+
+        return str_contains($contentType, "application/json") || str_contains($contentType, "multipart/form-data");
+    }
 }
```

Comment:
@jonwaldstein I think we should check the [Accept](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept) header instead and have the client use that header with `application/json`. Semantically, I don't think it's correct to assume that a `multipart/form-data` content request implies a JSON response. But the Accept header gives the client a way of explicitly specifying the response type.

Reply:
@JasonTheAdams ah yeah thanks for the suggestion.  This was kind of bothering me too beacuse I know that we don't want to necessarily _assume_ the response should be json if using `multipart/form-data` I just figured that these are the two content-types we are using that should return json and didn't think about the accept header.  

So, I think your suggestion is the better approach because it's more of an explicit way for the client to specify what it wants in return (by default the fetch api is literally anything `*/*`).  

To be clear we would do this on the client:
``` 
fetch(url, {
  headers: {
    accept: 'application/json'
  }
})
```
 and on the server check `$_SERVER['HTTP_ACCEPT']` or even  `getallheaders()["Accept"]` for json right?

Reply:
Updated here e9328f4b